### PR TITLE
current.getLong("id") to selected

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
## Purpose / Description
Improving ankidroid code quality. Shorter code and avoiding useless computation.

## Approach
selected() returns the id. Current is defined as get(selected()). So
doing current.getLong("id") takes the id, get the deck, and then takes
the id.

## How Has This Been Tested?
Put in my "merged" branch, which I run on my phone with all PR I made

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java [hunk 1]
```diff
@@ -1969,8 +1969,8 @@ private void restoreCollectionPreferences() {
             // Dynamic don't have review options; attempt to get deck-specific auto-advance options
             // but be prepared to go with all default if it's a dynamic deck
             JSONObject revOptions = new JSONObject();
-            if (!getCol().getDecks().isDyn(getCol().getDecks().current().getLong("id"))) {
-                revOptions = getCol().getDecks().confForDid(getCol().getDecks().current().getLong("id")).getJSONObject("rev");
+            if (!getCol().getDecks().isDyn(getCol().getDecks().selected())) {
+                grevOptions = getCol().getDecks().confForDid(getCol().getDecks().selected()).getJSONObject("rev");
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java:1973]: Seems fine, I just think 'grevOptions' was a typo and you meant the following?

```suggestion
                revOptions = getCol().getDecks().confForDid(getCol().getDecks().selected()).getJSONObject("rev");
```

File: AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java [hunk 2]
```diff
@@ -1969,8 +1969,9 @@ private void restoreCollectionPreferences() {
             // Dynamic don't have review options; attempt to get deck-specific auto-advance options
             // but be prepared to go with all default if it's a dynamic deck
             JSONObject revOptions = new JSONObject();
-            if (!getCol().getDecks().isDyn(getCol().getDecks().current().getLong("id"))) {
-                revOptions = getCol().getDecks().confForDid(getCol().getDecks().current().getLong("id")).getJSONObject("rev");
+            long selectedDid = getCol().getDecks().selected();
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.java:1972]: Codacy Issue found: [Avoid unused local variables such as 'selectedDid'.](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5330111)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5330111&bid=17445320)

**Comment by mikehardy** [comment]
This released as 2.10alpha65 as soon as google is done processing it

**Pull request merged**